bleak/__init__.py

```diff
@@ -26,63 +47,518 @@
     _logger.addHandler(handler)
     _logger.setLevel(logging.DEBUG)
 
-if _on_rtd:
-    pass
-elif os.environ.get("P4A_BOOTSTRAP") is not None:
-    from bleak.backends.p4android.scanner import (
-        BleakScannerP4Android as BleakScanner,
-    )  # noqa: F401
-    from bleak.backends.p4android.client import (
-        BleakClientP4Android as BleakClient,
-    )  # noqa: F401
-elif platform.system() == "Linux":
-    from bleak.backends.bluezdbus.scanner import (
-        BleakScannerBlueZDBus as BleakScanner,
-    )  # noqa: F401
-    from bleak.backends.bluezdbus.client import (
-        BleakClientBlueZDBus as BleakClient,
-    )  # noqa: F401
-elif platform.system() == "Darwin":
-    try:
-        from CoreBluetooth import CBPeripheral  # noqa: F401
-    except Exception as ex:
-        raise BleakError("Bleak requires the CoreBluetooth Framework") from ex
-
-    from bleak.backends.corebluetooth.scanner import (
-        BleakScannerCoreBluetooth as BleakScanner,
-    )  # noqa: F401
-    from bleak.backends.corebluetooth.client import (
-        BleakClientCoreBluetooth as BleakClient,
-    )  # noqa: F401
-
-elif platform.system() == "Windows":
-    # Requires Windows 10 Creators update at least, i.e. Window 10.0.16299
-    _vtup = platform.win32_ver()[1].split(".")
-    if int(_vtup[0]) != 10:
-        raise BleakError(
-            "Only Windows 10 is supported. Detected was {0}".format(
-                platform.win32_ver()
-            )
+
+class BleakScanner:
+    """
+    Interface for Bleak Bluetooth LE Scanners.
+
+    Args:
+        detection_callback:
+            Optional function that will be called each time a device is
+            discovered or advertising data has changed.
+        service_uuids:
+            Optional list of service UUIDs to filter on. Only advertisements
+            containing this advertising data will be received. Required on
+            macOS >= 12.0, < 12.3 (unless you create an app with ``py2app``).
+        scanning_mode:
+            Set to ``"passive"`` to avoid the ``"active"`` scanning mode.
+            Passive scanning is not supported on macOS! Will raise
+            :class:`BleakError` if set to ``"passive"`` on macOS.
+        bluez:
+            Dictionary of arguments specific to the BlueZ backend.
+        **kwargs:
+            Additional args for backwards compatibility.
+    """
+
+    def __init__(
+        self,
+        detection_callback: Optional[AdvertisementDataCallback] = None,
+        service_uuids: Optional[List[str]] = None,
+        scanning_mode: Literal["active", "passive"] = "active",
+        *,
+        bluez: BlueZScannerArgs = {},
```

Comment:
importing `BlueZScannerArgs` could cause import errors on other platforms and should probably just be `dict` here.

Reply:
you can make it type checking only:

```
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .backends.bluezdbus.scanner import BlueZScannerArgs
```

Reply:
Nice idea. That would fix runtime errors for sure. We'll see how it affects docs and static analysis tools.

Reply:
it worked :+1: